cognee/modules/graph/utils/get_model_instance_from_graph.py

```diff
@@ -1,29 +1,56 @@
+from typing import Callable
+
 from pydantic_core import PydanticUndefined
+
 from cognee.infrastructure.engine import DataPoint
 from cognee.modules.storage.utils import copy_model
 
 
-def get_model_instance_from_graph(nodes: list[DataPoint], edges: list, entity_id: str):
-    node_map = {}
+def merge_dicts(dict1: dict, dict2: dict, agg_fn: Callable) -> dict:
+    merged_dict = {}
+    for key, value in dict1.items():
+        if key in dict2:
+            merged_dict[key] = agg_fn(value, dict2[key])
+        else:
+            merged_dict[key] = value
+
+    for key, value in dict2.items():
+        if key not in merged_dict:
+            merged_dict[key] = value
+    return merged_dict
+
 
-    for node in nodes:
-        node_map[node.id] = node
+def get_model_instance_from_graph(
+    nodes: list[DataPoint],
+    edges: list[tuple[str, str, str, dict[str, str]]],
+    entity_id: str,
+):
+    node_map = {node.id: node for node in nodes}
 
-    for edge in edges:
-        source_node = node_map[edge[0]]
-        target_node = node_map[edge[1]]
-        edge_label = edge[2]
-        edge_properties = edge[3] if len(edge) == 4 else {}
+    for source_node_id, target_node_id, edge_label, edge_properties in edges:
+        source_node = node_map[source_node_id]
+        target_node = node_map[target_node_id]
         edge_metadata = edge_properties.get("metadata", {})
-        edge_type = edge_metadata.get("type")
+        edge_type = edge_metadata.get("type", "default")
 
         if edge_type == "list":
-            NewModel = copy_model(type(source_node), { edge_label: (list[type(target_node)], PydanticUndefined) })
-
-            node_map[edge[0]] = NewModel(**source_node.model_dump(), **{ edge_label: [target_node] })
+            NewModel = copy_model(
+                type(source_node),
+                {edge_label: (list[type(target_node)], PydanticUndefined)},
+            )
+            new_model_dict = merge_dicts(
```

Comment:
`merge_dicts` seems like an overkill here, we want to merge just one property here. It will iterate over properties of both objects for one property, right?

Reply:
Yes, good point, I changed it to merge the key values directly.